### PR TITLE
Remove dataset policy when deleting dataset

### PR DESCRIPTION
**Pull request by tenyo**

### CODE REVIEW

**Comment by damnski** [comment]
lgtm

File: s3datarepository/iam.go [hunk 1]
```diff
@@ -177,8 +177,40 @@ func (s *S3Repository) deletePolicy(ctx context.Context, id string) error {
...
+		EntityFilter:      aws.String("Role"),
+		PathPrefix:        aws.String(s.IAMPathPrefix),
+		PolicyArn:         aws.String(policyArn),
+		PolicyUsageFilter: aws.String("PermissionsPolicy"),
+	})
+	if err != nil {
+		log.Warnf("failed to list entities for policy "+policyArn, err)
+	}
+
+	if entitiesOut != nil {
```
Comment (fishnix) [inline s3datarepository/iam.go:196]: I don't think this is every nil unless there's an error

Comment (fishnix) [reply s3datarepository/iam.go:196]: oh i see you don't return above if there's an error...

File: s3datarepository/iam.go [hunk 2]
```diff
@@ -177,8 +177,40 @@ func (s *S3Repository) deletePolicy(ctx context.Context, id string) error {
...
+		log.Debug(entitiesOut.PolicyRoles)
+		if len(entitiesOut.PolicyRoles) == 0 {
+			log.Debugf("policy %s is not attached to any roles", policyName)
+		}
```
Comment (fishnix) [inline s3datarepository/iam.go:200]: this log statement conditional seems excessive... 😆 

`log.Debugf("policy %s is attached to %d roles", policyName, len(entitiesOut.PolicyRoles))`

**Pull request merged**